### PR TITLE
python311Packages.distrax: 0.1.4 -> 0.1.5

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

`distrax` is currently broken on `master`.
This PR updates it to the latest version and disables the broken tests.
(Those failures have also been reported upstream).

Changelog: https://github.com/google-deepmind/distrax/releases/tag/v0.1.5

cc @onny

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 268921` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.distrax</li>
    <li>python310Packages.distrax.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
`rlax` does not build because its dependency `plotnine` is broken (unrelated to this change).

**Comment by natsukium** [comment]
We need to disable these three tests.

- JointTest.test_nested_distributions__with_device
- JointTest.test_nested_distributions__without_device
- JointTest.test_nested_distributions__without_jit
<details><summary>error log on aarch64-linux</summary>

```
=================================== FAILURES ===================================
_______________ JointTest.test_nested_distributions__with_device _______________
self = <distrax._src.distributions.joint_test.JointTest testMethod=test_nested_distributions__with_device>
    @chex.all_variants
    def test_nested_distributions(self):
      distributions, inputs = _make_nested_distributions_and_inputs()
      joint = Joint(distributions)
    
      key = jax.random.PRNGKey(0)
      subkeys = jax.random.split(key, 6)
    
      with self.subTest('sample'):
        actuals = self.variant(joint.sample)(seed=key)
        assert isinstance(actuals, dict)
        assert isinstance(actuals['categoricals'], list)
        assert isinstance(actuals['normals'], tuple)
        assert isinstance(actuals['multivariate'], jnp.ndarray)
    
        flat_actuals = tree.flatten(actuals)
        flat_dists = tree.flatten(distributions)
        for actual, dist, subkey in zip(flat_actuals, flat_dists, subkeys):
          expected = dist.sample(seed=subkey)
          np.testing.assert_allclose(actual, expected, rtol=1e-6)
    
      with self.subTest('log_prob'):
        actual = self.variant(joint.log_prob)(inputs)
        flat_dists = tree.flatten(distributions)
        flat_inputs = tree.flatten(inputs)
        log_probs = [dist.log_prob(x) for dist, x in zip(flat_dists, flat_inputs)]
        expected = sum(log_probs)
>       np.testing.assert_array_equal(actual, expected)
distrax/_src/distributions/joint_test.py:269: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
args = (<built-in function eq>, Array(-16.8070382, dtype=float64), Array(-16.8070382, dtype=float64))
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}
    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Arrays are not equal
E           
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 3.55271368e-15
E           Max relative difference: 2.11382496e-16
E            x: array(-16.807038)
E            y: array(-16.807038)
/nix/store/7ahkv87jj59z90yal5dcrgagz58cqmz6-python3-3.11.6/lib/python3.11/contextlib.py:81: AssertionError
_____________ JointTest.test_nested_distributions__without_device ______________
self = <distrax._src.distributions.joint_test.JointTest testMethod=test_nested_distributions__without_device>
    @chex.all_variants
    def test_nested_distributions(self):
      distributions, inputs = _make_nested_distributions_and_inputs()
      joint = Joint(distributions)
    
      key = jax.random.PRNGKey(0)
      subkeys = jax.random.split(key, 6)
    
      with self.subTest('sample'):
        actuals = self.variant(joint.sample)(seed=key)
        assert isinstance(actuals, dict)
        assert isinstance(actuals['categoricals'], list)
        assert isinstance(actuals['normals'], tuple)
        assert isinstance(actuals['multivariate'], jnp.ndarray)
    
        flat_actuals = tree.flatten(actuals)
        flat_dists = tree.flatten(distributions)
        for actual, dist, subkey in zip(flat_actuals, flat_dists, subkeys):
          expected = dist.sample(seed=subkey)
          np.testing.assert_allclose(actual, expected, rtol=1e-6)
    
      with self.subTest('log_prob'):
        actual = self.variant(joint.log_prob)(inputs)
        flat_dists = tree.flatten(distributions)
        flat_inputs = tree.flatten(inputs)
        log_probs = [dist.log_prob(x) for dist, x in zip(flat_dists, flat_inputs)]
        expected = sum(log_probs)
>       np.testing.assert_array_equal(actual, expected)
distrax/_src/distributions/joint_test.py:269: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
args = (<built-in function eq>, Array(-16.8070382, dtype=float64), Array(-16.8070382, dtype=float64))
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}
    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Arrays are not equal
E           
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 3.55271368e-15
E           Max relative difference: 2.11382496e-16
E            x: array(-16.807038)
E            y: array(-16.807038)
/nix/store/7ahkv87jj59z90yal5dcrgagz58cqmz6-python3-3.11.6/lib/python3.11/contextlib.py:81: AssertionError
_______________ JointTest.test_nested_distributions__without_jit _______________
self = <distrax._src.distributions.joint_test.JointTest testMethod=test_nested_distributions__without_jit>
    @chex.all_variants
    def test_nested_distributions(self):
      distributions, inputs = _make_nested_distributions_and_inputs()
      joint = Joint(distributions)
    
      key = jax.random.PRNGKey(0)
      subkeys = jax.random.split(key, 6)
    
      with self.subTest('sample'):
        actuals = self.variant(joint.sample)(seed=key)
        assert isinstance(actuals, dict)
        assert isinstance(actuals['categoricals'], list)
        assert isinstance(actuals['normals'], tuple)
        assert isinstance(actuals['multivariate'], jnp.ndarray)
    
        flat_actuals = tree.flatten(actuals)
        flat_dists = tree.flatten(distributions)
        for actual, dist, subkey in zip(flat_actuals, flat_dists, subkeys):
          expected = dist.sample(seed=subkey)
          np.testing.assert_allclose(actual, expected, rtol=1e-6)
    
      with self.subTest('log_prob'):
        actual = self.variant(joint.log_prob)(inputs)
        flat_dists = tree.flatten(distributions)
        flat_inputs = tree.flatten(inputs)
        log_probs = [dist.log_prob(x) for dist, x in zip(flat_dists, flat_inputs)]
        expected = sum(log_probs)
>       np.testing.assert_array_equal(actual, expected)
distrax/_src/distributions/joint_test.py:269: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
args = (<built-in function eq>, Array(-16.8070382, dtype=float64), Array(-16.8070382, dtype=float64))
kwds = {'err_msg': '', 'header': 'Arrays are not equal', 'strict': False, 'verbose': True}
    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Arrays are not equal
E           
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 3.55271368e-15
E           Max relative difference: 2.11382496e-16
E            x: array(-16.807038)
E            y: array(-16.807038)
/nix/store/7ahkv87jj59z90yal5dcrgagz58cqmz6-python3-3.11.6/lib/python3.11/contextlib.py:81: AssertionError
```
</details>

**Comment by GaetanLepage** [comment]
> We need to disable these three tests.

Done, and by looking at the "pattern", I also disabled the `with_jit` variant as well.

File: pkgs/development/python-modules/distrax/default.nix [hunk 1]
```diff
@@ -34,6 +40,26 @@ buildPythonPackage rec {
     "distrax"
   ];
 
+  disabledTests = [
+    # AssertionError on numerical values
+    # Reported upstream in https://github.com/google-deepmind/distrax/issues/267
```
Comment (samuela) [inline pkgs/development/python-modules/distrax/default.nix:45]: ❤️

**Review by samuela — APPROVED** [review]
diff LGTM

**Comment by natsukium** [comment]
@ofborg eval

**Comment by GaetanLepage** [comment]
> @ofborg eval

ofborg seems happy !

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**